llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -1710,20 +1711,19 @@ void SampleProfileLoader::generateMDProfMetadata(Function &F) {
           else if (OverwriteExistingWeights)
             I.setMetadata(LLVMContext::MD_prof, nullptr);
         } else if (!isa<IntrinsicInst>(&I)) {
-          I.setMetadata(LLVMContext::MD_prof,
-                        MDB.createBranchWeights(
-                            {static_cast<uint32_t>(BlockWeights[BB])}));
+          setBranchWeights(I, {static_cast<uint32_t>(BlockWeights[BB])});
         }
       }
     } else if (OverwriteExistingWeights || ProfileSampleBlockAccurate) {
       // Set profile metadata (possibly annotated by LTO prelink) to zero or
       // clear it for cold code.
       for (auto &I : *BB) {
         if (isa<CallInst>(I) || isa<InvokeInst>(I)) {
-          if (cast<CallBase>(I).isIndirectCall())
+          if (cast<CallBase>(I).isIndirectCall()) {
             I.setMetadata(LLVMContext::MD_prof, nullptr);
```

Comment:
Didn't mean to keep asking for more, but the inconsistency in nearby code wrt dealing with setMetadata directly, vs calling setBranchWeights seem not ideal. I would add clearBranchWeights too. But not gonna block the patch because of that.

Reply:
Sorry this line was part of PR https://github.com/llvm/llvm-project/pull/72446 that I already landed. Hope it's fine to leave this unchanged.